js/woo.js

```diff
@@ -0,0 +1,1465 @@
+        //         {
+        //             protocol: "ERC20",
+        //             token: "USDT",
+        //             name: "Ethereum",
+        //             minimum_withdrawal: 30,
+        //             withdrawal_fee: 25,
+        //             allow_deposit: 1,
+        //             allow_withdraw: 1
+        //         },
+        //         {
+        //             protocol: "TRC20",
+        //             token: "USDT",
+        //             name: "Tron",
+        //             minimum_withdrawal: 30,
+        //             withdrawal_fee: 1,
+        //             allow_deposit: 1,
+        //             allow_withdraw: 1
+        //         },
+        //         ...
+        //     ],
+        //     success: true
+        // }
+        //
+        const derivedNetworksData = {};
+        for (let i = 0; i < data2.length; i++) {
+            const item = data2[i];
+            const tokenCode = this.safeString (item, 'token');
+            const code = this.safeCurrencyCode (tokenCode);
+            if (!(code in derivedNetworksData)) {
+                derivedNetworksData[code] = {
+                    'networks': {},
+                };
+            }
+            const networkId = this.detectChainSlug (this.safeString (item, 'protocol', ''));
+            derivedNetworksData[code]['networks'][networkId] = {
+                'network_title': this.safeString (item, 'name'),
+                'allow_deposit': this.safeString (item, 'allow_deposit'),
+                'allow_withdraw': this.safeString (item, 'allow_withdraw'),
+                'withdrawal_fee': this.safeString (item, 'withdrawal_fee'),
+                'minimum_withdrawal': this.safeString (item, 'minimum_withdrawal'),
+                'info': { 'networkInfo': item },
+            };
+        }
+        // combine as final step.
+        const keys = Object.keys (derivedCurrenciesData); // keys and items amount in derivedCurrenciesData and derivedNetworksData are same
+        for (let i = 0; i < keys.length; i++) {
+            const code = keys[i]; // keys are alraedy safeCurrencyCode-d
+            const currencyInfo = derivedCurrenciesData[code];
+            const currencyNetworks = derivedNetworksData[code];
+            const currencyMerged = this.deepExtend (currencyInfo, currencyNetworks);
+            // if response1 each item contains only one child, then they are the same network . Quite bad behavior of their API for this matter, but as-is: the only solution is this as I think at this moment.
+            const name = this.safeString (currencyMerged, 'fullname');
+            const isActive = this.hasChildWithKeyValue (currencyNetworks, 'allow_deposit', 1) && this.hasChildWithKeyValue (currencyNetworks, 'allow_withdraw', 1);
+            result[code] = {
+                'id': code,
+                'name': name,
+                'code': code,
+                'precision': undefined, // TODO: different networks have different precision
```

Comment:
See how that is done here:

- https://github.com/ccxt/ccxt/blob/master/js/okex.js#L940
- https://github.com/ccxt/ccxt/blob/master/js/huobi.js#L2237